src/util/dynamicfilesequencedownloader.cpp

```diff
@@ -0,0 +1,657 @@
+        }
+        if (!success) {
+            if (attempt < maxRetries) {
+                LINFO(std::format("Retry number {}.", attempt + 1));
+                std::this_thread::sleep_for(std::chrono::seconds(2));
+                response.start();
+                response.wait();
+            }
+            else {
+                LERROR(std::format(
+                    "Failed according to warning above with http request of url: {}",
+                    httpInfoRequest
+                ));
+            }
+        }
+        attempt++;
+    }
+}
+
+void DynamicFileSequenceDownloader::requestAvailableFiles(std::string httpDataRequest,
+                                                          std::filesystem::path syncDir)
+{
+    // Declaring big window. pair.first is timestep, pair.second is url to be downloaded
+
+    // Get big window/ list of available files for the specified data set.
+    // If it expands to more of a API call rather than a http-request, that code goes here
+    // TODO: make_unique?
+    HttpMemoryDownload response = HttpMemoryDownload(httpDataRequest);
+    response.start();
+    response.wait();
+
+    bool success = false;
+    int attempt = 0;
+    const int maxRetries = 1;
+    nlohmann::json jsonResult;
+
+    /********************
+    *   Example response
+    *********************
+    * {
+    *  "dataID": 1234,
+    *  "files": [
+    *   {
+    *    "timestamp": "2017-07-01 00:42:02.0",
+    *    "url": "https://iswa...fieldlines/trace_scs_outtoin/timestamp.osfls"
+    *   },
+    *   {
+    *    "timestamp": "2017-07-01 00:51:36.0",
+    *    "url": "https://iswa...fieldlines/trace_scs_outtoin/timestamp.osfls"
+    *   }
+    *  ],
+    *  "time.max": "2017-10-01 00:00:00.0",
+    *  "time.min": "2017-06-01 00:00:00.0"
+    * }
+    ********************
+    * note that requested time can be month 10 but last entry in list is month 07,
+    * meaning there are no more available files between month 7-10.
+    * *****************/
+    while (attempt <= maxRetries && !success) {
+        try {
+            std::vector<char> data = response.downloadedData();
+            if (data.empty()) {
+                throw ghoul::RuntimeError("Empty HTTP response");
+            }
+            //TODO what value is actually to large to handle?
+            if (data.size() > std::numeric_limits<std::size_t>::max()) {
+                throw ghoul::RuntimeError(
+                    "Http responds with list of available files to large"
+                );
+            }
+
+            jsonResult = nlohmann::json::parse(data);
+            success = true;
+        }
+        catch (const nlohmann::json::parse_error& ex) {
+            LERROR(std::format("JSON parsing error: '{}'", ex.what()));
+        }
+        catch (const std::bad_alloc& ex) {
+            LERROR(std::format(
+                "Memory allocation error while parsing JSON: '{}'",
+                ex.what()
+            ));
+        }
```

Comment:
```suggestion
```

There is barely any reason ever to catch a `bad_alloc` since we're out of memory at that point anyway